app/src/main/java/com/paypal/psix/services/UserSession.java

```diff
@@ -50,13 +56,26 @@ public static void setUser(GraphUser newUser) {
         new UserSession(new User(newUser.getId(), newUser.getFirstName(), newUser.getLastName()));
     }
 
-    public static void userLoggedOut() {
+    public static void logout(Activity activity) {
+        new Delete().from(Event.class).execute();
```

Comment:
Wouldn't it be enough to delete the User, and then all its events and the event's rsvps will be deleted?